### DISCUSSION TITLE
USWDS 3.0.0 Beta 1

### DISCUSSION THREAD

**Discussion by thisisdano** [uswds/uswds#4386] [General]
## What's new in 3.0.0 Beta 1

This release is up-to-date with USWDS 2.12.2 and incorporates more of our new directory structure and some of the Pattern Lab → Storybook migration.

### What's in this release
- [x] Drop support for IE11 [IN PROGRESS]
- [x] Reorganize internal directory structure [IN PROGRESS]
- [ ] ~Use Sass Module Syntax~ [Coming in a later release]
- [ ] ~Publish component packages~ [Coming in a later release]

## Updating from stable
Install on your project with `npm install --save uswds@beta`

If you're using `includePaths` in sass, you'll need to update the path from `./node_modules/uswds/dist/scss` to ` /node_modules/uswds/dist/scss/stylesheets`.

Example using USWDS-Gulp:
```js
// OLD
.pipe(
  sass({
    includePaths: [
      `sass/styles.scss`,
      `${uswds}/scss`,
      `${uswds}/scss/packages`,
    ],
  })
)

// NEW
.pipe(
  sass({
    includePaths: [
      `sass/styles.scss`,
      `${uswds}/scss/stylesheets`,
      `${uswds}/scss/stylesheets/packages`,
    ],
  })
)
```

### ⚠️ Breaking changes

**Updated package names**
Only affects those using USWDS packages. The required and global partials are now prefixed with `uswds-`.

For example:
```scss
@import 'packages/required';
@import 'packages/global';
```

Required and global SCSS packages have been updated to:
```scss
@import 'packages/uswds-required';
@import 'packages/uswds-global';
```

## Security 

0 vulnerabilities in regular dependencies (dependencies for USWDS projects installed with npm install uswds)

Internal only: 93 vulnerabilities (26 moderate, 65 high, 2 critical) in devDependencies (development dependencies)

Release ZIP SHA-256 hash: 4226f63ef09f7064bff4e169b3bda8d94f1da2ce866592302bca7b84bf91018f

<hr /><em>This discussion was created from the release <a href='https://github.com/uswds/uswds/releases/tag/v3.0.0-beta.1'>USWDS 3.0.0 Beta 1</a>.</em>

**Comment by mbucknell** [comment]
My first attempt to use USWDS 3.0 beta in an existing project (code.usgs.gov/waterdataui). A couple of differences besides the location of the style sheets. 

- If you are using dynamic javascript and importing USWDS components the path has changed and the organization is different. I had to change from:
```javascript
import {tooltip as uswds_tooltip} from 'uswds-components';// Where uswds-components was mapped to the uswds/src/components directory in node_modules
```
to:
```javascript
import uswds_tooltip from 'uswds-components/usa-tooltip/tooltip'
```
I use buble to compile the javascript and had to add the transform, dangerousTaggedTemplateString, and set it to true. One of the other components we use in the date range picker and I think this was the issue.

I have not dug in deeply to see what the issue is but my radio buttons and checkbox labels are appearing partial behind the button:
<img width="379" alt="Screen Shot 2021-11-22 at 11 32 54 AM" src="https://user-images.githubusercontent.com/8104736/142908307-9fde146a-19e4-43d2-8dc7-33d530924a30.png">

Otherwise, I haven't found any major issues. Please note that we build our style sheets with gulp and use rollup/buble for the javascript. The original code repo is: https://code.usgs.gov/wma/iow/waterdataui and and example production page is (using uswds 2.12.2: https://waterdata.usgs.gov/monitoring-location/05429369/#parameterCode=00065&period=P7D

**Reply by mbucknell** [reply]
Regarding the radio button, checkbox label styling, it looks like the style for both the usa-radio__label and usa-checkbox__label include setting the text-indent attribute to:
```css
text-indent: units(-$input-select-margin-right - $theme-input-select-size);
```
Using Chrome developer tools in the CSS inspector, if a disable the text-indent setting, the labels appear where I expect them.

**Reply by mejiaj** [reply]
Thanks for the feedback! The radio button is a regression that's been fixed and we hope to release soon.
